Project/core/src/main/java/Controller/GameController.java

```diff
@@ -0,0 +1,118 @@
+package Controller;
+
+import Model.Player;
+import Model.Recruiter;
+import Model.RougeAgent;
+import Model.Game;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GameController {
+    // Check for new turn
+    // Think this would be better as view calling it to move from one turn to the
+    // next as needed
+    // Update recruits if odd turn
+    // Activate player
+    // Check win
+
+    private boolean recruiterTurn = true;
+    private Game gameState;
+    private int activePlayer = 0;
+    private Model.Player playerTurnOrder[] = new Player[6];
+    private Recruiter recruiter = null;
+
+    /**
+     * The main gameController. Keeps an eye on victory conditions and which players
+     * are next in queue to play. This is the constructor. Currently does not track
+     * when
+     * recruits and similar have been added timewise.
+     * 
+     * @param newGame A game model object meant to describe everything for view
+     */
+
+    public GameController(Game newGame) {
+        // Create turn order
+        // This controller will use this to know which player controls what unit
+        int agentIterator = 0; // This is in case there are less than four agents. Every unit will still be
+                               // controlled
+
+        gameState = newGame;
+        List<Player> gamePlayers = gameState.getPlayers();
+        List<RougeAgent> agents = new ArrayList<RougeAgent>();
+
+        boolean oneRecruiter = true;
+        for (int i = 0; i < gamePlayers.size(); i++) {
+            Player currPlayer = gamePlayers.get(i);
+            if (currPlayer instanceof Recruiter) {
+                recruiter = (Recruiter) currPlayer;
+                gameState.setCurrentPlayer(recruiter);
+                if (!oneRecruiter) {
+                    throw new IllegalStateException("More than one recruiter");
+                }
+                oneRecruiter = false;
+            } else {
+                agents.add((RougeAgent) currPlayer);
+            }
+        }
+
+        if (recruiter == null) {
+            throw new IllegalStateException("Recruiter player not found");
+        }
+        if (agents.isEmpty()) {
+            throw new IllegalStateException("No agents found");
+        }
+
+        for (int i = 0; i < playerTurnOrder.length; i++) {
+            switch (i) {
+                case 0:
+                    playerTurnOrder[i] = recruiter;
+                    break;
+                case 3:
+                    playerTurnOrder[i] = recruiter;
+                    break;
+                default:
+                    playerTurnOrder[i] = agents.get(agentIterator);
+                    agentIterator++;
+                    if (agentIterator >= agents.size()) {
+                        agentIterator = 0;
+                    }
+            }
+        }
```

Comment:
This playerTurnOrder array only works for some cases, but not for all. For example, if we have 1 recruiter and 3 agents, the array won't let the agens play equally often. I suggest we let it be and mark it as an issue which we correct sometime in the future.